src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx

```diff
@@ -20,6 +20,512 @@ export function getStaticProps(context) {
+	await confirmSignIn({
+		challengeResponse: 'EMAIL', // 'EMAIL', 'SMS', or 'TOTP'
+	});
+}
+
+if (nextStep.signInStep === 'CONTINUE_SIGN_IN_WITH_MFA_SETUP_SELECTION') {
+	// present nextStep.allowedMFATypes to user
+	// collect user selection
+	await confirmSignIn({
+		challengeResponse: 'EMAIL', // 'EMAIL' or 'TOTP'
+	});
+}
+
+if (nextStep.signInStep === 'CONTINUE_SIGN_IN_WITH_EMAIL_SETUP') {
+	// collect email address from user
+	await confirmSignIn({
+		challengeResponse: 'hello@mycompany.com',
+	});
+}
+
+if (nextStep.signInStep === 'CONTINUE_SIGN_IN_WITH_TOTP_SETUP') {
+	// present nextStep.totpSetupDetails.getSetupUri() to user
+	// collect OTP from user
+	await confirmSignIn({
+		challengeResponse: '123456',
+	});
+}
+
+if (nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE') {
+	// collect custom challenge answer from user
+	await confirmSignIn({
+		challengeResponse: 'custom-challenge-answer',
+	});
+}
+
+if (nextStep.signInStep === 'CONFIRM_SIGN_IN_WITH_NEW_PASSWORD_REQUIRED') {
+	// collect new password from user
+	await confirmSignIn({
+		challengeResponse: 'new-password',
+	});
+}
+
+if (nextStep.signInStep === 'RESET_PASSWORD') {
+	// initiate reset password flow
+	await resetPassword({
+		username: 'username',
+	});
+}
+
+if (nextStep.signInStep === 'CONFIRM_SIGN_UP') {
+	// user was not confirmed during sign up process
+	// if user has confirmation code, invoke `confirmSignUp` api
+	// otherwise, invoke `resendSignUpCode` to resend the code
+	await confirmSignUp({
+		username: 'username',
+		confirmationCode: '123456',
+	});
+}
+
+if (nextStep.signInStep === 'DONE') {
+	// signin complete
+}
+```
+
+## Confirm signin with SMS MFA
+
+If the next step is `CONFIRM_SIGN_IN_WITH_SMS_CODE`, Amplify Auth has sent the user a random code over SMS and is waiting for the user to verify that code. To handle this step, your app's UI must prompt the user to enter the code. After the user enters the code, pass the value to the `confirmSignIn` API.
+
+<Callout>
+
+The result includes `codeDeliveryDetails` with additional information about the code delivery, such as the partial phone number of the SMS recipient, which can be used to prompt the user on where to look for the code.
+
+</Callout>
+
+```ts
+async function handleSignInResult(result: SignInOutput) {
+	switch (result.nextStep.signInStep) {
+		case 'CONFIRM_SIGN_IN_WITH_SMS_CODE': {
+			const { codeDeliveryDetails } = result.nextStep;
+			// OTP has been delivered to user via SMS
+			// Inspect codeDeliveryDetails for additional delivery information
+			console.log(
+				`A confirmation code has been sent to ${codeDeliveryDetails?.destination}`,
+			);
+			console.log(
+				`Please check your ${codeDeliveryDetails?.deliveryMedium} for the code.`,
+			);
+			break;
+		}
+	}
+}
+
+```
+
+```ts
+async function confirmMfaCode(mfaCode: string) {
+	const result = await confirmSignIn({ challengeResponse: mfaCode });
+
+	return handleSignInResult(result);
+}
+
```

Comment:
this can be one snippet like the other page

Reply:
[consolidated snippets here](https://github.com/aws-amplify/docs/pull/8023/commits/ee45f7c6aa0523c892ae574977c079ce62326106)